### DISCUSSION TITLE
Extending hydra configs from multiple directories with hydra.searchpath

### DISCUSSION THREAD

**Discussion by dorotat-nv** [facebookresearch/hydra#2717] [Q&A]
Hi! In my use cases, we have some dependencies between hydra configs for training different models - different config groups.

However, the configs may be located in different directories. I am wondering how to tackle it without the limitation of hydra.searchpath being allowed only in the primary config. Ie an example

`config /A/conf/A.yaml` - some base config
`config /B/conf/B.yaml `that uses as a default `/A/conf/A.yaml` with `hydra.search.path = "/A/conf"`
`config /B/conf/B1.yaml` that uses as a default `/B/conf/B.yaml` from the same folder
`config /C/conf/C.yaml` that uses as a default `/B/conf/B.yaml` with `hydra.search.path = "/B/conf"`
Hydra will process with no problems configs in (1) and (2) but throws errors at (3) and (4) since they contain inherited hydra.searchpath from the defaults configurations.

Ideally, I would like to come up with a solution that attaches hydra.searchpaths from the non-primary configs to the hydra.searchpath as this approach gives the most visibility to where the configs are located

**Comment by odelalleau** [comment]
I can't think of a really good solution to this problem -- besides making Hydra support setting `hydra.searchpath` in non-primary configs so as to dynamically update the search path during composition (but I have no idea how hard it would be to implement this).

Here are two potential workarounds:
1. Provide `hydra.searchpath` on the command-line (must contain all paths required to load all parent configs)
2. Create two versions of each config: one with `hydra.searchpath` (used when it is the primary config -- should list all search paths of parents as well) and one without (used when included from child configs)

None of these workarounds is really satisfying since they are somewhat cumbersome and require updating either the command line (option 1) or the config (option 2) for all children of a config whose search path is updated.

Edit: Note that in option 2 above you wouldn't duplicate the config, instead the config with `hydra.searchpath` would include the other one through the defaults list
